### PR TITLE
feat(fpm-writer): property `allowAutoAddDependencyLib` to the building block API, which allows turning off the automatic addition of the 'sap.fe.macros' library

### PR DESCRIPTION
**Pull request by 815are**
Change:
1. `fpm-writer`. Addition property `allowAutoAddDependencyLib` to the building block API, which allows turning off the automatic addition of the 'sap.fe.macros' library
2. `fpm-writer` - return code snippet for `manifest.json` based on `allowAutoAddDependencyLib`
3. `examples/prompting-ui` - updated example with option to toggle `allowAutoAddDependencyLib`:
Option ON:
image
Option OFF:
image

### CODE REVIEW

**Review by AlinaGovoruhina — APPROVED** [review]
Added property to switch adding library to manifest, unit test coverage good

File: .changeset/sweet-rivers-dress.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@sap-ux/prompting-ui': patch
+---
```
Comment (devinea) [inline .changeset/sweet-rivers-dress.md:3]: not sure this changeset is needed when `examples/prompting-ui`is updated.

**Review by devinea — APPROVED** [review]
code changes are clear and covered by tests

File: .changeset/sweet-rivers-dress.md (hunk 1 above)
Comment (815are) [reply .changeset/sweet-rivers-dress.md:3]: I am also was not sure, but saw that other example packages also has changeset file like
https://github.com/SAP/open-ux-tools/blob/main/examples/odata-cli/CHANGELOG.md
-> https://github.com/SAP/open-ux-tools/pull/2157/files#diff-4d640317d13b1ce460db0d741e3049ada8421ba4f0657fdd1f1d28eb6e0b836aR3

**Pull request merged**